sequence_task_1/student/include/impl.h

```diff
@@ -8,33 +8,105 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using value_type = typename Container::value_type;
+    using allocator_type = typename Container::allocator_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using pointer = typename Container::pointer;
+    using const_pointer = typename Container::const_pointer;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using size_type = typename Container::size_type;
+    using difference_type = typename Container::difference_type;
 
     /** @todo Iterators */
+    const_iterator begin() const {
+        return m_tracklist.cbegin();
+    }
 
-    StaticPlaylist() = default;
+    const_iterator end() const {
+        return m_tracklist.cend();
+    }
+
+    const_iterator rbegin() const {
+        return m_tracklist.rbegin();
+    }
+
+    const_iterator rend() const {
+        return m_tracklist.rend();
+    }
+
+    const_iterator cbegin() const
+    {
+        return m_tracklist.cbegin();
+    }
+
+    const_iterator cend() const
+    {
+        return m_tracklist.cend();
+    }
+
+    const_iterator crbegin() const
+    {
+        return m_tracklist.crbegin();
+    }
+
+    const_iterator crend() const
+    {
+        return m_tracklist.crend();
+    }
+
+    StaticPlaylist() : m_tracklist() {}
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c80-use-default-if-you-have-to-be-explicit-about-using-the-default-semantics